File: backend/models/employee.py
from .db import BasicModel

class Employee(BasicModel):
    def __init__(self):
        super().__init__(model='employees')

    def get_list(self, date=None):
        employees = self.collection.find(sort=[('name', 1)])
        employees_list = []
        for employee in employees:
            # 퇴사하지 않은 직원만 포함
            if employee['regular'] != '퇴사':            
                if date:
                    if (date >= employee['beginDate'] and 'endDate' not in employee) or ('endDate' in employee and date <= employee['endDate']):
                        employees_list.append(employee)
                else:
                    employees_list.append(employee)
        return employees_list

    def get(self, employee_id=None):
        return self.collection.find_one({'employeeId': employee_id})

    def post(self, employee_id=None, name=None, begin_date=None):
        if employee_id and name and begin_date:
            employee = self.get(employee_id=employee_id)
            if not employee:
                self.collection.insert_one({'employeeId': employee_id, 'name': name, 'beginDate': begin_date, 'regular': '상근', 'mode': '내근', 'attendMode': 'X'})

    def get_approver(self, employee):
        approver = None
        if employee['position'] == '팀원':
            approver = self.collection.find_one({'position': '팀장', 'department': employee['department'], 'regular': {'$ne': '퇴사'}})
            if not approver:
                approver = self.collection.find_one({'position': '본부장', 'regular': {'$ne': '퇴사'}})
        elif employee['position'] == '팀장':
            approver = self.collection.find_one({'position': '본부장', 'regular': {'$ne': '퇴사'}})
        elif employee['position'] == '본부장':
            approver = employee
        elif employee['position'] == '대표이사':
            approver = employee
        else:
            approver = self.collection.find_one({'position': '본부장', 'regular': {'$ne': '퇴사'}})
        return approver

File: api/controllers/employee.js
import Employee from '../models/Employee.js'
import User from '../models/User.js'
import { sanitizeData } from '../utils/util.js'
import EmployeeService from '../services/EmployeeService.js'

/**
 * Controller for Employee management
 */

export const search = async (req, res, next) => {
    try {
        const employees = await getEmployees(req)
        res.status(200).setHeader('csrftoken', req.csrfToken()).json(employees)
    } catch (err) {
        next(err)
    }
}

export const update = async (req, res, next) => {
    try {
        const { _id, beginDate: rawBeginDate, email, mobileNo: mobileNoStr, department, rank, position, regular, mode, attendMode, cardNo } = req.body
        const beginDate = sanitizeData(rawBeginDate, 'date')
        const mobileNo = sanitizeData(mobileNoStr, 'mobile')
        await Employee.updateOne({ _id }, { $set: { beginDate, email, mobileNo, department, rank, position, regular, mode, attendMode, cardNo } }, { runValidators: true })
        
        // 퇴사하는 경우 User 계정도 삭제 
        if (regular === EmployeeService.STATUS.RETIRED) {
            await User.deleteOne({ email })
        }
        res.status(204).send()
    } catch (err) {
        next(err)
    }
}

export const write = async (req, res, next) => {
    try {
        const { employeeId: rawEmployeeId, name, beginDate: rawBeginDate, email, department, rank, position, regular, mode, attendMode, cardNo } = req.body
        const employeeId = Number(rawEmployeeId)
        const beginDate = sanitizeData(rawBeginDate, 'date')
        const newEmployee = new Employee({ employeeId, name, beginDate, email, department, rank, position, regular, mode, attendMode, cardNo })
        await newEmployee.save()
        res.status(200).json(newEmployee)
    } catch (err) {
        next(err)
    }
}

export const deleteEmployee = async (req, res, next) => {
    try {
        const { _id } = req.body
        const employee = await Employee.deleteOne({ _id })
        res.status(200).json(employee)
    } catch (err) {
        next(err)
    }
}

const getEmployees = async (req) => {
    const name = req.query.name
    const isAdmin = req.user.isAdmin
    
    let query = {}
    if (name) query.name = name

    // 관리자가 아닌 경우 cardNo 필드 제외 (기존 로직 유지)
    if (isAdmin) {
        return await EmployeeService.getActiveEmployees(query)
    } else {
        return await Employee.find({ ...query, regular: { $ne: EmployeeService.STATUS.RETIRED } }, { cardNo: 0 }).sort({ name: 1 })
    }
}

/**
 * Proxy for legacy internal calls or other services
 */
export const getEmployeeByEmail = async (email) => {
    return await EmployeeService.validateActiveEmployee(email).then(res => res.employee)
}